dissect/target/plugins/os/unix/config.py

```diff
@@ -81,54 +105,216 @@ def parse_file(self, fh: io.TextIO) -> None:
+
+    def _parse_line(self, line: str, data_dict: dict) -> tuple:
+        key, *values = self.SEPERATOR.split(line)
+        values = " ".join(value for value in values if value).strip()
+
+        if old_value := data_dict.get(key):
+            if not isinstance(old_value, list):
+                old_value = [old_value]
+            values = old_value + [values]
+        data_dict[key] = values
+
+        return key
+
+
+@dataclass(frozen=True)
+class ParserContents:
+    collapse: Optional[Union[bool, set]] = None
+    seperator: Optional[tuple[str]] = None
+    comment_prefixes: Optional[tuple[str]] = None
+
+
+@dataclass(frozen=True)
+class ParserConfig:
+    parser: type[ConfigurationParser] = Default
+    collapse: Optional[Union[bool, set]] = None
+    seperator: Optional[tuple[str]] = None
+    comment_prefixes: Optional[tuple[str]] = None
+
+    def create_parser(self, contents: Optional[ParserContents] = None) -> ConfigurationParser:
+        kwargs = {}
+
+        for field_name in ["collapse", "seperator", "comment_prefixes"]:
+            value = getattr(contents, field_name, None) or getattr(self, field_name)
+            if value:
+                kwargs.update({field_name: value})
+
+        return self.parser(**kwargs)
+
+
+CONFIG_MAP: dict[str, ParserConfig] = {
+    "ini": ParserConfig(Ini),
+    "xml": ParserConfig(Txt),
+    "json": ParserConfig(Txt),
+    "cnf": ParserConfig(Default),
+    "conf": ParserConfig(Default, seperator=(r"\s")),
+    "sample": ParserConfig(Txt),
+    "template": ParserConfig(Txt),
 }
-KNOWN_FILES: dict[str, LinuxConfigurationParser] = {
-    "ulogd.conf": Ini,
+KNOWN_FILES: dict[str, type[ConfigurationParser]] = {
+    "ulogd.conf": ParserConfig(Ini),
+    "hosts.allow": ParserConfig(Default, seperator=(":",), comment_prefixes=("#",)),
+    "hosts.deny": ParserConfig(Default, seperator=(":",), comment_prefixes=("#",)),
+    "hosts": ParserConfig(Default, seperator=(r"\s")),
 }
 
 
-class ConfigurationFs(VirtualFilesystem):
-    __fstype__: str = "META:registry"
+def parse_config(
+    entry: FilesystemEntry,
+    hint: Optional[str] = None,
+    contents: Optional[ParserContents] = None,
+) -> ConfigParser:
+    parser_type = _select_parser(entry, hint)
+
+    parser = parser_type.create_parser(contents)
+
+    with entry.open() as fh:
+        open_file = io.TextIOWrapper(fh, encoding="utf-8")
+        parser.read_file(open_file)
+
+    return parser
+
+
+def _select_parser(entry: FilesystemEntry, hint: Optional[str] = None) -> ParserConfig:
+    if hint and (parser_type := CONFIG_MAP.get(hint)):
+        return parser_type
 
-    def __init__(self, target: Target, **kwargs):
+    extension = entry.path.rsplit(".", 1)[-1]
+
+    known_extension = CONFIG_MAP.get(extension, ParserConfig(Default))
+    return KNOWN_FILES.get(entry.name, known_extension)
+
+
+def create_entry(
+    fs, path: str, entry, hint: str, collapse: str, seperator: tuple[str], comment_prefixes: tuple[str]
```

Comment:
Probably nicer to unroll all the parameters. Also `entry` is missing a type hint.